Index: app/src/main/res/layout/app_bar_main.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/app_bar_main.xml	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/res/layout/app_bar_main.xml	(revision )
@@ -22,12 +22,12 @@
 
     <include layout="@layout/content_main" />
 
-    <android.support.design.widget.FloatingActionButton
-        android:id="@+id/fab"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_gravity="bottom|end"
-        android:layout_margin="@dimen/fab_margin"
-        app:srcCompat="@drawable/ic_power_settings_new" />
+    <!--<android.support.design.widget.FloatingActionButton-->
+        <!--android:id="@+id/fab"-->
+        <!--android:layout_width="wrap_content"-->
+        <!--android:layout_height="wrap_content"-->
+        <!--android:layout_gravity="bottom|end"-->
+        <!--android:layout_margin="@dimen/fab_margin"-->
+        <!--app:srcCompat="@drawable/ic_power_settings_new" />-->
 
 </android.support.design.widget.CoordinatorLayout>
Index: app/src/main/res/values/strings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/strings.xml	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/res/values/strings.xml	(revision )
@@ -142,5 +142,10 @@
     <string name="reason">Reason</string>
     <string name="add_role">Add Role</string>
     <string name="add_user">Add User</string>
+    <string name="data">Data</string>
+    <string name="receive">Receive</string>
+    <string name="export">Export</string>
+    <string name="logout">Logout</string>
+    <string name="today_s_revenue">Today\'s Revenue</string>
 
 </resources>
Index: app/src/main/java/com/mitrakreasindo/pos/main/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/mitrakreasindo/pos/main/MainActivity.java	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/java/com/mitrakreasindo/pos/main/MainActivity.java	(revision )
@@ -10,6 +10,7 @@
 import android.support.v7.app.ActionBarDrawerToggle;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
+import android.view.Menu;
 import android.view.MenuItem;
 import android.view.View;
 import android.widget.TextView;
@@ -20,6 +21,7 @@
 import com.mitrakreasindo.pos.common.TableHelper.TablePeopleHelper;
 import com.mitrakreasindo.pos.common.TableHelper.TableProductHelper;
 import com.mitrakreasindo.pos.common.TableHelper.TableRoleHelper;
+import com.mitrakreasindo.pos.main.fragment.MainFragment;
 import com.mitrakreasindo.pos.main.fragment.MaintenanceFragment;
 import com.mitrakreasindo.pos.main.fragment.SalesFragment;
 import com.mitrakreasindo.pos.main.fragment.StockFragment;
@@ -29,72 +31,88 @@
 public class MainActivity extends AppCompatActivity
         implements NavigationView.OnNavigationItemSelectedListener
 {
-    private String valueUser;
-    private int mPrevSelectedId;
-    private String companyCode;
+  private String valueUser;
+  private int mPrevSelectedId;
+  private String companyCode;
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState)
-    {
-        super.onCreate(savedInstanceState);
+  @Override
+  protected void onCreate(Bundle savedInstanceState)
+  {
+    super.onCreate(savedInstanceState);
 
-        setContentView(R.layout.activity_main);
-        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
-        setSupportActionBar(toolbar);
+    setContentView(R.layout.activity_main);
+    Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+    setSupportActionBar(toolbar);
 
-        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
-        fab.setOnClickListener(new View.OnClickListener()
-        {
-            @Override
-            public void onClick(View view) {
-                startActivity(new Intent(MainActivity.this, RoleActivity.class));
-//                Logout();
-//                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
-//                        .setAction("Action", null).show();
-            }
-        });
+//    FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
+//    fab.setOnClickListener(new View.OnClickListener()
+//    {
+//      @Override
+//      public void onClick(View view)
+//      {
+//        startActivity(new Intent(MainActivity.this, RoleActivity.class));
+////                Logout();
+////                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
+////                        .setAction("Action", null).show();
+//      }
+//    });
 
-        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
-        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
-                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
-        drawer.setDrawerListener(toggle);
-        toggle.syncState();
+    DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+    ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
+            this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
+    drawer.setDrawerListener(toggle);
+    toggle.syncState();
 
-        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
-        navigationView.setNavigationItemSelectedListener(this);
+    NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
+    navigationView.setNavigationItemSelectedListener(this);
 
-        valueUser = getIntent().getExtras().getString("USERNAME");
-        Toast.makeText(this, "Welcome " + valueUser, Toast.LENGTH_SHORT).show();
-        Toast.makeText(this, "Current Schema: " +
-                getIntent().getExtras().getString("COMPANY"), Toast.LENGTH_SHORT).show();
+    valueUser = getIntent().getExtras().getString("USERNAME");
+    Toast.makeText(this, "Welcome " + valueUser, Toast.LENGTH_SHORT).show();
+    Toast.makeText(this, "Current Schema: " +
+            getIntent().getExtras().getString("COMPANY"), Toast.LENGTH_SHORT).show();
 
-        View headerLayout = navigationView.getHeaderView(0);
-        TextView textViewUser = (TextView) headerLayout.findViewById(R.id.textViewUser);
-        textViewUser.setText(valueUser);
+    View headerLayout = navigationView.getHeaderView(0);
+    TextView textViewUser = (TextView) headerLayout.findViewById(R.id.textViewUser);
+    textViewUser.setText(valueUser);
 
-        companyCode = SharedPreferenceEditor.LoadPreferences(this, "");
+    companyCode = SharedPreferenceEditor.LoadPreferences(this, "");
 
+    FragmentManager fragmentManager = getSupportFragmentManager();
+    MainFragment mainFragment = new MainFragment();
+    fragmentManager.beginTransaction().replace(R.id.main_content, mainFragment, mainFragment.getTag()).commit();
+
-        TablePeopleHelper tablePeopleHelper = new TablePeopleHelper(this);
-        tablePeopleHelper.downloadData(companyCode);
-        TableRoleHelper tableRoleHelper = new TableRoleHelper(this);
-        tableRoleHelper.downloadData(companyCode);
-        TableCategoryHelper tableCategoryHelper = new TableCategoryHelper(this);
-        tableCategoryHelper.downloadData(companyCode);
-        TableProductHelper tableProductHelper = new TableProductHelper(this);
-        tableProductHelper.downloadDataAlternate();
-    }
+    TablePeopleHelper tablePeopleHelper = new TablePeopleHelper(this);
+    tablePeopleHelper.downloadData(companyCode);
+    TableRoleHelper tableRoleHelper = new TableRoleHelper(this);
+    tableRoleHelper.downloadData(companyCode);
+    TableCategoryHelper tableCategoryHelper = new TableCategoryHelper(this);
+    tableCategoryHelper.downloadData(companyCode);
+    TableProductHelper tableProductHelper = new TableProductHelper(this);
+    tableProductHelper.downloadDataAlternate();
+  }
 
-    @Override
+  @Override
-    public void onBackPressed() {
+  public void onBackPressed()
+  {
-        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+    DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
-        if (drawer.isDrawerOpen(GravityCompat.START)) {
+    if (drawer.isDrawerOpen(GravityCompat.START))
+    {
-            drawer.closeDrawer(GravityCompat.START);
+      drawer.closeDrawer(GravityCompat.START);
-        } else {
+    }
+    else
+    {
-            super.onBackPressed();
-        }
-    }
+      super.onBackPressed();
+    }
+  }
 
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu)
+  {
+    getMenuInflater().inflate(R.menu.dashboard_menu, menu);
+    return super.onCreateOptionsMenu(menu);
+  }
+
-//    @Override
+  //    @Override
 //    public boolean onCreateOptionsMenu(Menu menu)
 //    {
 //        // Inflate the menu; this adds items to the action bar if it is present.
@@ -102,71 +120,73 @@
 //        return true;
 //    }
 
-    @Override
-    public boolean onOptionsItemSelected(MenuItem item)
-    {
-        // Handle action bar item clicks here. The action bar will
-        // automatically handle clicks on the Home/Up button, so long
-        // as you specify a parent activity in AndroidManifest.xml.
-        int id = item.getItemId();
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item)
+  {
+    // Handle action bar item clicks here. The action bar will
+    // automatically handle clicks on the Home/Up button, so long
+    // as you specify a parent activity in AndroidManifest.xml.
+    int id = item.getItemId();
 
-        //noinspection SimplifiableIfStatement
+    //noinspection SimplifiableIfStatement
-        if (id == R.id.action_settings) {
-            return true;
+    if (id == R.id.action_logout)
+    {
+      Toast.makeText(this, "Goodbye " + valueUser + ". See you next time :)", Toast.LENGTH_SHORT).show();
+      finish();
-        }
+    }
 
-        return super.onOptionsItemSelected(item);
-    }
+    return super.onOptionsItemSelected(item);
+  }
 
-    @SuppressWarnings("StatementWithEmptyBody")
-    @Override
-    public boolean onNavigationItemSelected(MenuItem item)
-    {
-        // Handle navigation view item clicks here.
-        int id = item.getItemId();
+  @SuppressWarnings("StatementWithEmptyBody")
+  @Override
+  public boolean onNavigationItemSelected(MenuItem item)
+  {
+    // Handle navigation view item clicks here.
+    int id = item.getItemId();
 
-        if (id == R.id.nd_logout)
-        {
-            Logout();
-        }
-        else if (id == R.id.nd_maintenance)
-        {
-            Toast.makeText(this, "Maintenance", Toast.LENGTH_LONG).show();
-            getSupportActionBar().setTitle("Maintenance");
-            MaintenanceFragment maintenanceFragment = new MaintenanceFragment();
-            FragmentManager fragmentManager = getSupportFragmentManager();
-            fragmentManager.beginTransaction().replace(R.id.main_content, maintenanceFragment, maintenanceFragment.getTag()).commit();
-        }
-        else if (id == R.id.nd_stock)
-        {
-            Toast.makeText(this, "Stock", Toast.LENGTH_LONG).show();
-            getSupportActionBar().setTitle("Stock");
-            StockFragment stockFragment = new StockFragment();
-            FragmentManager fragmentManager = getSupportFragmentManager();
-            fragmentManager.beginTransaction().replace(R.id.main_content, stockFragment, stockFragment.getTag()).commit();
-        }
-        else if (id == R.id.nd_sales)
-        {
-            Toast.makeText(this, "Sales", Toast.LENGTH_LONG).show();
-            getSupportActionBar().setTitle("Sales");
-            SalesFragment salesFragment = new SalesFragment();
-            FragmentManager fragmentManager = getSupportFragmentManager();
-            fragmentManager.beginTransaction().replace(R.id.main_content, salesFragment, salesFragment.getTag()).commit();
-        }
-        else if (id == R.id.nd_customer_payment)
-        {
-            Toast.makeText(this, "Product", Toast.LENGTH_LONG).show();
-            startActivity(new Intent(this, ProductFormActivity.class));
-        }
+    if (id == R.id.nd_logout)
+    {
+      Logout();
+    }
+    else if (id == R.id.nd_maintenance)
+    {
+      Toast.makeText(this, "Maintenance", Toast.LENGTH_LONG).show();
+      getSupportActionBar().setTitle("Maintenance");
+      MaintenanceFragment maintenanceFragment = new MaintenanceFragment();
+      FragmentManager fragmentManager = getSupportFragmentManager();
+      fragmentManager.beginTransaction().replace(R.id.main_content, maintenanceFragment, maintenanceFragment.getTag()).commit();
+    }
+    else if (id == R.id.nd_stock)
+    {
+      Toast.makeText(this, "Stock", Toast.LENGTH_LONG).show();
+      getSupportActionBar().setTitle("Stock");
+      StockFragment stockFragment = new StockFragment();
+      FragmentManager fragmentManager = getSupportFragmentManager();
+      fragmentManager.beginTransaction().replace(R.id.main_content, stockFragment, stockFragment.getTag()).commit();
+    }
+    else if (id == R.id.nd_sales)
+    {
+      Toast.makeText(this, "Sales", Toast.LENGTH_LONG).show();
+      getSupportActionBar().setTitle("Sales");
+      SalesFragment salesFragment = new SalesFragment();
+      FragmentManager fragmentManager = getSupportFragmentManager();
+      fragmentManager.beginTransaction().replace(R.id.main_content, salesFragment, salesFragment.getTag()).commit();
+    }
+    else if (id == R.id.nd_customer_payment)
+    {
+      Toast.makeText(this, "Product", Toast.LENGTH_LONG).show();
+      startActivity(new Intent(this, ProductFormActivity.class));
+    }
 
-        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
-        drawer.closeDrawer(GravityCompat.START);
-        return true;
-    }
+    DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+    drawer.closeDrawer(GravityCompat.START);
+    return true;
+  }
 
-    private void Logout()
-    {
+  private void Logout()
+  {
-        Toast.makeText(this, "Goodbye "+ valueUser + ". See you next time :)", Toast.LENGTH_SHORT).show();
+    Toast.makeText(this, "Goodbye " + valueUser + ". See you next time :)", Toast.LENGTH_SHORT).show();
-        finish();
-    }
+    finish();
+  }
 }
Index: app/src/main/java/com/mitrakreasindo/pos/main/stock/product/ProductActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/mitrakreasindo/pos/main/stock/product/ProductActivity.java	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/java/com/mitrakreasindo/pos/main/stock/product/ProductActivity.java	(revision )
@@ -14,6 +14,7 @@
 import android.widget.TextView;
 
 import com.mitrakreasindo.pos.common.ClientService;
+import com.mitrakreasindo.pos.common.SharedPreferenceEditor;
 import com.mitrakreasindo.pos.common.TableHelper.TableProductHelper;
 import com.mitrakreasindo.pos.main.R;
 import com.mitrakreasindo.pos.main.stock.product.controller.ProductListAdapter;
@@ -47,6 +48,7 @@
   private Product product;
   private TableProductHelper tableProductHelper;
 
+//  private SharedPreferenceEditor sharedPreferenceEditor;
   public boolean is_action_mode = false;
 
   @Override
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/build.gradle	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/build.gradle	(revision )
@@ -47,6 +47,7 @@
     compile 'com.squareup.retrofit2:retrofit:2.3.0'
     compile 'com.squareup.retrofit2:converter-jackson:2.3.0'
     compile 'com.readystatesoftware.sqliteasset:sqliteassethelper:+'
+    compile 'de.codecrafters.tableview:tableview:2.6.0'
     testCompile 'junit:junit:4.12'
     annotationProcessor 'com.jakewharton:butterknife-compiler:8.5.1'
 }
Index: app/src/main/res/values/colors.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/colors.xml	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/res/values/colors.xml	(revision )
@@ -10,5 +10,13 @@
     <color name="gray400">#BDBDBD</color>
     <color name="gray500">#9E9E9E</color>
     <color name="white">#FFF</color>
+
     <color name="red500">#F44336</color>
+    <color name="red400">#F44336</color>
+
+    <color name="teal">#1DE9B6</color>
+    <color name="light_green">#00E676</color>
+    <color name="blue">#4FC3F7</color>
+    <color name="blue_gray">#90A4AE</color>
+    <color name="deep_purple">#7E57C2</color>
 </resources>
Index: app/src/main/java/com/mitrakreasindo/pos/main/stock/category/CategoryFormActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/mitrakreasindo/pos/main/stock/category/CategoryFormActivity.java	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/java/com/mitrakreasindo/pos/main/stock/category/CategoryFormActivity.java	(revision )
@@ -63,6 +63,8 @@
 
     categoryListAdapter = new CategoryListAdapter(this, new ArrayList<Category>());
 
+    sharedPreferenceEditor = new SharedPreferenceEditor();
+
     kodeMerchant = sharedPreferenceEditor.LoadPreferences(this, "");
 
     bundle = getIntent().getExtras();
\ No newline at end of file
Index: app/src/main/java/com/mitrakreasindo/pos/main/maintenance/user/UserActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/mitrakreasindo/pos/main/maintenance/user/UserActivity.java	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/java/com/mitrakreasindo/pos/main/maintenance/user/UserActivity.java	(revision )
@@ -14,6 +14,7 @@
 import android.text.TextWatcher;
 import android.view.View;
 import android.widget.EditText;
+import android.widget.LinearLayout;
 import android.widget.Toast;
 
 import com.mitrakreasindo.pos.common.ClientService;
@@ -43,7 +44,7 @@
   @BindView(R.id.list_users)
   RecyclerView listUsers;
   @BindView(R.id.main_content)
-  ConstraintLayout mainContent;
+  LinearLayout mainContent;
   @BindView(R.id.fab_users)
   FloatingActionButton fabUsers;
 
Index: app/src/main/java/com/mitrakreasindo/pos/main/stock/category/CategoryActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/mitrakreasindo/pos/main/stock/category/CategoryActivity.java	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/java/com/mitrakreasindo/pos/main/stock/category/CategoryActivity.java	(revision )
@@ -2,16 +2,17 @@
 
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.v4.view.MenuItemCompat;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
-import android.support.v7.widget.SearchView;
 import android.support.v7.widget.Toolbar;
+import android.text.Editable;
+import android.text.TextWatcher;
 import android.util.Log;
 import android.view.Menu;
 import android.view.MenuItem;
 import android.view.View;
+import android.widget.EditText;
 import android.widget.LinearLayout;
 
 import com.mitrakreasindo.pos.common.ClientService;
@@ -39,6 +40,8 @@
   RecyclerView listCategory;
   @BindView(R.id.main_content)
   LinearLayout mainContent;
+  @BindView(R.id.edit_text_filter)
+  EditText editTextFilter;
 
   private CategoryService categoryService;
   private CategoryListAdapter categoryListAdapter;
@@ -66,6 +69,8 @@
       }
     });
 
+    sharedPreferenceEditor = new SharedPreferenceEditor();
+
     kodeMerchant = sharedPreferenceEditor.LoadPreferences(this, "");
 
     categoryService = ClientService.createService().create(CategoryService.class);
@@ -79,6 +84,27 @@
 
     tableCategoryHelper = new TableCategoryHelper(this);
 
+    editTextFilter.addTextChangedListener(new TextWatcher()
+    {
+      @Override
+      public void onTextChanged(CharSequence s, int start, int before, int count)
+      {
+        categoryListAdapter.clear();
+        categoryListAdapter.addCategory(tableCategoryHelper.getData(editTextFilter.getText().toString()));
+      }
+
+      @Override
+      public void beforeTextChanged(CharSequence s, int start, int count, int after)
+      {
+
+      }
+
+      @Override
+      public void afterTextChanged(Editable s)
+      {
+
+      }
+    });
     categoryListAdapter.clear();
     categoryListAdapter.addCategory(tableCategoryHelper.getData());
 
Index: app/src/main/java/com/mitrakreasindo/pos/main/maintenance/role/RoleActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/mitrakreasindo/pos/main/maintenance/role/RoleActivity.java	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/java/com/mitrakreasindo/pos/main/maintenance/role/RoleActivity.java	(revision )
@@ -43,7 +43,7 @@
   Toolbar toolbar;
   @BindView(R.id.list_role)
   RecyclerView listRole;
-  @BindView(R.id.edittext_filter)
+  @BindView(R.id.edit_text_filter)
   EditText editTextFilter;
 
   private RoleAdapter roleAdapter;
@@ -51,6 +51,7 @@
   private Role role;
   private String kodeMerchant;
   private TableRoleHelper tableRoleHelper;
+  private SharedPreferenceEditor sharedPreferenceEditor;
 
   @Override
   protected void onCreate(@Nullable Bundle savedInstanceState)
@@ -70,7 +71,9 @@
       }
     });
 
-    kodeMerchant = SharedPreferenceEditor.LoadPreferences(this, "");
+    sharedPreferenceEditor = new SharedPreferenceEditor();
+
+    kodeMerchant = sharedPreferenceEditor.LoadPreferences(this, "");
 
     roleService = ClientService.createService().create(RoleService.class);
     roleAdapter = new RoleAdapter(this, new ArrayList<Role>());
Index: app/src/main/java/com/mitrakreasindo/pos/main/maintenance/user/UserFormActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/mitrakreasindo/pos/main/maintenance/user/UserFormActivity.java	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/java/com/mitrakreasindo/pos/main/maintenance/user/UserFormActivity.java	(revision )
@@ -111,6 +111,10 @@
     peopleService = ClientService.createService().create(PeopleService.class);
 
     bundle = getIntent().getExtras();
+    if (bundle != null)
+    {
+
+    }
     String peopleId = bundle.getString("id");
     final String name = bundle.getString("name");
     final String password = bundle.getString("password");
\ No newline at end of file
Index: app/src/main/java/com/mitrakreasindo/pos/service/ProductService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/mitrakreasindo/pos/service/ProductService.java	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/java/com/mitrakreasindo/pos/service/ProductService.java	(revision )
@@ -16,4 +16,7 @@
 {
   @GET("chromis.products/{kodeMerchant}/")
   Call<List<Product>> getProductAll(@Path("kodeMerchant") String kodeMerchant);
+
+  @GET("chromis.products/{kodeMerchant}/{id}")
+  Call<List<Product>> getProductById(@Path("kodeMerchant") String kodeMerchant, @Path("id") String id);
 }
Index: app/src/main/java/com/mitrakreasindo/pos/main/maintenance/role/RoleFormActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/mitrakreasindo/pos/main/maintenance/role/RoleFormActivity.java	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/java/com/mitrakreasindo/pos/main/maintenance/role/RoleFormActivity.java	(revision )
@@ -42,6 +42,7 @@
 
   private Role role;
   private RoleService roleService;
+  private SharedPreferenceEditor sharedPreferenceEditor;
   private String kodeMerchant;
 
   @Override
@@ -63,7 +64,9 @@
 
     roleService = ClientService.createService().create(RoleService.class);
 
-    kodeMerchant = SharedPreferenceEditor.LoadPreferences(this, "");
+    sharedPreferenceEditor = new SharedPreferenceEditor();
+
+    kodeMerchant = sharedPreferenceEditor.LoadPreferences(this, "");
 
     final Bundle bundle = getIntent().getExtras();
 
Index: app/src/main/res/layout/product_divider.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/product_divider.xml	(revision 6280b4c9941e924c71527f6d154c0229daeec36b)
+++ app/src/main/res/layout/product_divider.xml	(revision )
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <View android:background="@color/gray300"
       android:layout_width="match_parent"
-      android:layout_height="1dp"
+      android:layout_height="0.5dp"
       xmlns:android="http://schemas.android.com/apk/res/android" />
\ No newline at end of file
